rules/order/src/Rector/ClassMethod/OrderConstructorDependenciesByTypeAlphabeticallyRector.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Order\Rector\ClassMethod;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt\ClassMethod;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\Order\Tests\Rector\ClassMethod\OrderConstructorDependenciesByTypeAlphabeticallyRector\OrderConstructorDependenciesByTypeAlphabeticallyRectorTest
+ */
+final class OrderConstructorDependenciesByTypeAlphabeticallyRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Order __constructor dependencies by type A-Z', [
+            new CodeSample(
+                <<<'PHP'
+class SomeClass
+{
+    public function __construct(
+        LatteToTwigConverter $latteToTwigConverter,
+        SymfonyStyle $symfonyStyle,
+        LatteAndTwigFinder $latteAndTwigFinder,
+        SmartFileSystem $smartFileSystem,
+    ) {
+    }
+}
+PHP
+,
+                <<<'PHP'
+class SomeClass
+{
+    public function __construct(
+        LatteAndTwigFinder $latteAndTwigFinder,
+        LatteToTwigConverter $latteToTwigConverter,
+        SmartFileSystem $smartFileSystem,
+        SymfonyStyle $symfonyStyle,
+    ) {
+    }
+}
+PHP
+
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        $node->params = $this->getSortedParams($node);
+
+        return $node;
+    }
+
+    /**
+     * @return Param[]
+     */
+    private function getSortedParams(ClassMethod $classMethod): array
+    {
+        $params = $classMethod->getParams();
+        usort($params, function (Param $a, Param $b) {
```

Comment:
```suggestion
        usort($params, function (Param $firstParam, Param $secondParam) {
```

Names should respect types as normal params do.

Possible place for next naming rule ;)